### ISSUE TITLE
Amend PDF download feature

### ISSUE THREAD

**Issue by monicamow** [agenda-item/agenda-item#88]
- currently, the PDF includes squashed text and shows '|object, Object|' in the printout for missing users

**Comment by jenreiher** [comment]
If I don't already have a pull request for the Object, object thing I can make one - I just didn't want to fix right before we demoed!

**Comment by jenreiher** [comment]
Ok removing Object. object did not help the formatting - @monicamow can I leave this one with you?

**Comment by monicamow** [comment]
@jenreiher yes, this is my feature and I will OWN it. note to self: might be the css spacing or the way vue is squashing the agenda items when it loops through them

**Comment by monicamow** [comment]
OK I think I figured out what the issue is but I don't know how to fix it... 😖

**Comment by jenreiher** [comment]
What is it?

**Comment by monicamow** [comment]
The js plugin (xeponline/css2pdf/cloudformatter) is not really compatible with vue.js. The plugin works in a way where it targets the IDs of HTML elements. So it's able to target the meeting header (details) and footer (closing) when I gave it a specific ID since these are not contained inside a loop. But when I try to assign an ID to the div containing all the agenda items, it squashes them so I suspect it has something to do with how vue.js spits these elements onto the DOM. It also might have to do with the fact that the AI's are all output in to one div and so the plugin shoves them into one page - but this may be incorrect since I even tried increasing the size of the pdf page from letter to legal and it still squashes. So my suspicion is that it's treating the AI div with all the agenda items like some weird squashed section, instead of multiple little sections. I know this is a bad explanation 👎. I even tried giving each agenda item type its own ID and targeting that but the plugin is unable to target any ID that is created by vue.js since it's not directly in the HTML code until it gets sent to the browser. Anyways, my long-term solution is to explore jsPDF, a different plugin that I suspect doesn't rely on ID targeting, and instead handles pdf exporting based on the view of the DOM. My hope is that it acts as a pseudo-screenshot type thing.

The reason the plugin *appeared* to be working before was that the agenda items had that big movers and seconders table (see an old branch with meetings.erb) which was forcing each agenda item to be on a single page.

**Comment by monicamow** [comment]
Ruby solution: https://github.com/pdfkit/pdfkit
Rails solution: https://github.com/mileszs/wicked_pdf
